### PR TITLE
Updated documentation related to apt-cacher-ng

### LINKED ISSUE #11556: Test the apt-cache-ng example
Test this example and ensure sit is still viable. File any issues you find as `writing` issues.

http://docs.docker.com/examples/apt-cacher-ng/

### PR DESCRIPTION
**Pull request by Akasurde**
Fixes #11556

Signed-off-by: Abhijeet Kasurde akasurde@redhat.com

### CODE REVIEW

**Comment by thaJeztah** [comment]
Thanks @Akasurde. Did you have a look at @moxiegirl's suggestion to use the new "docker networks" feature? https://github.com/docker/docker/pull/18955#discussion_r48628247

**Comment by Akasurde** [comment]
@thaJeztah I am not quite familiar with "Docker Networks" concept. Can you given a hint or an example so that I can add that into documentation?

**Comment by thaJeztah** [comment]
The alternative for docker networks would be to create a custom network, and add your containers to that network. Containers can then reach each other through their name. Something like;

```bash
$ docker network create mynetwork

$ docker run -d -p 3142:3142 --net=mynetwork --name test_apt_cacher_ng eg_apt_cacher_ng

$docker run --rm -it --net=mynetwork -e http_proxy=http://test_apt_cacher_ng:3142/ debian bash
```

File: docs/examples/apt-cacher-ng.md [hunk 1]
```diff
@@ -53,7 +55,7 @@ use:
     $ docker logs -f test_apt_cacher_ng
 
 To get your Debian-based containers to use the proxy, you can do one of
```
Comment (vdemeester) [inline docs/examples/apt-cacher-ng.md]: `do one of the five things` feels a liitle bit weird ? -> `You have the following options` (I think we don't even need to count them :stuck_out_tongue_winking_eye:).

File: docs/examples/apt-cacher-ng.md [hunk 2]
```diff
@@ -30,6 +30,8 @@ Use the following Dockerfile:
     # and then you can run containers with:
     #   docker run -t -i --rm -e http_proxy http://dockerhost:3142/ debian bash
     #
+    # Here, `dockerhost` is the IP address or FQDN of a host running docker daemon
```
Comment (vdemeester) [inline docs/examples/apt-cacher-ng.md]: @moxiegirl @thaJeztah thinking out loud here, but do we have support for abbreviation on the documentation ? (`abbr` : https://developer.mozilla.org/en-US/docs/Web/HTML/Element/abbr)

**Comment by vdemeester** [comment]
Looks good, a tiny tiny comment though.

File: docs/examples/apt-cacher-ng.md (hunk 2 above)
Comment (jamtur01) [inline docs/examples/apt-cacher-ng.md]: running the Docker daemon.

File: docs/examples/apt-cacher-ng.md [hunk 3]
```diff
@@ -61,6 +63,9 @@ three things
    `http_proxy=http://dockerhost:3142/`
 3. Change your `sources.list` entries to start with
    `http://dockerhost:3142/`
+4. Link Debian-based containers to the APT proxy container using `--link`
+5. Create a custom network and add APT proxy container with Debian-based containers
```
Comment (jamtur01) [inline docs/examples/apt-cacher-ng.md]: Create a custom network and add an APT proxy container

Comment (jamtur01) [inline docs/examples/apt-cacher-ng.md]: Although perhaps that whole sentence needs a refactor?

File: docs/examples/apt-cacher-ng.md [hunk 4]
```diff
@@ -80,6 +85,16 @@ which obey `http_proxy`, such as `curl`, `wget` and others:
 might need to do it and you can do it from your `Dockerfile`
 too.
 
+**Option 4** is to link Debian-containers to proxy server using following command:
```
Comment (jamtur01) [inline docs/examples/apt-cacher-ng.md]: Sentence doesn't make sense.

File: docs/examples/apt-cacher-ng.md [hunk 5]
```diff
@@ -80,6 +85,16 @@ which obey `http_proxy`, such as `curl`, `wget` and others:
...
+
+    $ docker run -i -t --link test_apt_cacher_ng:apt_proxy -e http_proxy=http://apt_proxy:3142/ debian bash
+
+**Option 5** is to create a custom network of APT proxy server and Debian-based containers:
```
Comment (jamtur01) [inline docs/examples/apt-cacher-ng.md]: Sentence doesn't make sense.

    **Option 5** creates a custom network of APT proxy servers and Debian-based containers

**Comment by jamtur01** [comment]
Docs need some touch ups. Thanks!

**Comment by calavera** [comment]
LGTM

**Comment by vdemeester** [comment]
LGTM 🐧

**Pull request merged**